File: ejerc.diccionarios/Ejercicio 8/Ejercicio8.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class DiccionarioTraduccion {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in); // Para leer entradas del usuario
        Map<String, String> diccionario = new HashMap<>(); // Diccionario español-inglés

        // Crear diccionario
        System.out.print("Ingrese palabras en formato español:inglés separadas por comas: ");
        String entrada = sc.nextLine(); // Leer toda la línea
        String[] pares = entrada.split(","); // Separar por coma

        for (String par : pares) {
            String[] partes = par.trim().split(":"); // Separar por :
            if (partes.length == 2) { // Validar que tenga formato correcto
                diccionario.put(partes[0].trim(), partes[1].trim()); // Guardar en el diccionario
            }
        }

        // Frase a traducir
        System.out.print("Ingrese una frase en español: ");
        String frase = sc.nextLine();
        String[] palabras = frase.split(" "); // Separar la frase en palabras

        // Mostrar traducción
        System.out.print("Frase traducida: ");
        for (String palabra : palabras) {
            // Si la palabra no está en el diccionario, se deja igual
            System.out.print(diccionario.getOrDefault(palabra, palabra) + " ");
        }

        sc.close(); // Cerrar el escáner
    }
}

File: ejerc.diccionarios/Ejercicio 8/Ejercicio8.py
# Solicitar lista de traducciones en formato "español:inglés" separadas por comas
entrada = input("Ingrese palabras en formato español:inglés separadas por comas: ")

# Crear un diccionario vacío para almacenar las traducciones
diccionario = {}

# Procesar cada par de palabras separado por comas
for par in entrada.split(","):
    # Dividir cada par en español e inglés usando el carácter ":"
    esp, ing = par.strip().split(":")
    # Guardar la traducción en el diccionario, eliminando espacios innecesarios
    diccionario[esp.strip()] = ing.strip()

# Solicitar al usuario una frase en español
frase = input("Ingrese una frase en español: ")

# Lista para ir guardando las palabras traducidas
traduccion = []

# Traducir palabra por palabra
for palabra in frase.split():
    # Usar .get() para obtener la traducción si existe, o dejar la palabra igual si no está en el diccionario
    traduccion.append(diccionario.get(palabra, palabra))

# Mostrar el resultado uniendo las palabras traducidas con espacios
print("Frase traducida:", " ".join(traduccion))
